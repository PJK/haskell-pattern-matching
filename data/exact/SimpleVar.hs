module SimpleVar where

data MyTrinary = A | B | C
fun :: MyTrinary -> MyTrinary -> MyTrinary
fun x A = A
fun B B = A

