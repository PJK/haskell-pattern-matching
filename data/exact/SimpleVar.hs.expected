{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            "fun",
            {
                "tag": "NonExhaustive",
                "contents": [
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "A",
                                []
                            ]
                        },
                        {
                            "tag": "VariablePattern",
                            "contents": "fresh1"
                        }
                    ],
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "B",
                                []
                            ]
                        },
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "A",
                                []
                            ]
                        }
                    ],
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "B",
                                []
                            ]
                        },
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "C",
                                []
                            ]
                        }
                    ],
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "C",
                                []
                            ]
                        },
                        {
                            "tag": "VariablePattern",
                            "contents": "fresh1"
                        }
                    ]
                ]
            }
        ]
    ]
}